### ISSUE TITLE
loss: nan

### ISSUE THREAD

**Issue by arjunrajanna** [delijati/pytorch-siamese#1]
Hey,

Good work. However maybe you're aware, with default hyperparameter settings, loss: nan.

**Comment by kvananth** [comment]
I get nan too with default hyperparameter. Would you mind telling me what parameters you choose?

**Comment by delijati** [comment]
I will try to take a look into it after ma vacation

**Comment by delijati** [comment]
Not perfect but LGTM:
```
Args: Namespace(batchsize=128, cuda=True, epoch=3, model='', no_cuda=False)
Train Epoch: 1 [0/108400 (0%)]	Loss: 0.224497	Took: 0.52
Train Epoch: 1 [16384/108400 (15%)]	Loss: 0.089757	Took: 22.93
Train Epoch: 1 [32768/108400 (30%)]	Loss: 0.061219	Took: 22.98
Train Epoch: 1 [49152/108400 (45%)]	Loss: 0.069422	Took: 25.06
Train Epoch: 1 [65536/108400 (60%)]	Loss: 0.055707	Took: 25.43
Train Epoch: 1 [81920/108400 (76%)]	Loss: 0.051833	Took: 25.84
Train Epoch: 1 [98304/108400 (91%)]	Loss: 0.054869	Took: 26.29
Train epoch: 1 	Took:165.37
Train Epoch: 2 [0/108400 (0%)]	Loss: 0.042918	Took: 0.23
Train Epoch: 2 [16384/108400 (15%)]	Loss: 0.034205	Took: 26.40
Train Epoch: 2 [32768/108400 (30%)]	Loss: 0.038130	Took: 26.42
Train Epoch: 2 [49152/108400 (45%)]	Loss: 0.036047	Took: 27.43
Train Epoch: 2 [65536/108400 (60%)]	Loss: 0.035470	Took: 27.30
Train Epoch: 2 [81920/108400 (76%)]	Loss: 0.033549	Took: 27.22
Train Epoch: 2 [98304/108400 (91%)]	Loss: 0.044540	Took: 27.37
Train epoch: 2 	Took:179.39
Train Epoch: 3 [0/108400 (0%)]	Loss: 0.028087	Took: 0.23
Train Epoch: 3 [16384/108400 (15%)]	Loss: 0.031086	Took: 27.32
Train Epoch: 3 [32768/108400 (30%)]	Loss: 0.038790	Took: 27.39
Train Epoch: 3 [49152/108400 (45%)]	Loss: 0.032834	Took: 27.41
Train Epoch: 3 [65536/108400 (60%)]	Loss: 0.038789	Took: 27.40
Train Epoch: 3 [81920/108400 (76%)]	Loss: 0.028260	Took: 27.40
Train Epoch: 3 [98304/108400 (91%)]	Loss: 0.032505	Took: 27.31
Train epoch: 3 	Took:181.51
```

**Comment by iamhankai** [comment]
I get nan too with default hyperparameter...

**Comment by delijati** [comment]
cant reproduce sorry ...

**Comment by Peratham** [comment]
Using python3.5 will solve the problem.
I got nans when I was using python2.7.
I think there may be some compatibility issues in the code.

**Comment by delijati** [comment]
@Peratham thanks i never tried `python2` for that project my current setup is:

```bash
Python 3.5.3 (default, Jan 19 2017, 14:11:04) 
[GCC 6.3.0 20170118] on linux
Type "help", "copyright", "credits" or "license" for more information.
>>> import torch
>>> torch.__version__
'0.2.0+47f999b'
```

I am building `pytorch` by my own from master as i have  a `cuda` 2.0 nvidia card.
```
...
01:00.0 VGA compatible controller: NVIDIA Corporation GF108M [NVS 5400M] (rev a1)
...
```

Build instructions:
```bash
$ cd pytorch
$ virtualenv env3
$ sudo ln -s /usr/lib/x86_64-linux-gnu/libcudart.so* /usr/lib64/
$ rm -rf torch/lib/build torch/lib/tmp_install
$ env3/bin/python setup.py clean
$ NO_DISTRIBUTED=1 CC=/usr/bin/gcc-4.9 LD_LIBRARY_PATH=torch:/usr/lib/x86_64-linux-gnu/ TORCH_CUDA_ARCH_LIST="2.0 3.0 3.5 5.2 6.0 6.1+PTX" env3/bin/python setup.py bdist_wheel
$ env3/bin/pip install dist/torch*.whl
```



```

**Comment by Peratham** [comment]
I tried adding a dot to **255 -> 255.** at lines 47-47 and 59-60 and it solves the problem.
Just an integer division compatibility, I guess.
[http://sebastianraschka.com/Articles/2014_python_2_3_key_diff.html#integer-division](
http://sebastianraschka.com/Articles/2014_python_2_3_key_diff.html#integer-division)
